swap/src/cli/api/tauri_bindings.rs

```diff
@@ -193,23 +217,20 @@ impl TauriHandle {
                 },
             };
 
-            let mut map = self.0.pending_approvals.lock().await;
-            if let Some(pending) = map.remove(&request_id) {
-                let event = if accepted {
-                    ApprovalRequest::Resolved {
-                        request_id: request_id.to_string(),
-                        details: pending.details,
-                    }
-                } else {
-                    ApprovalRequest::Rejected {
-                        request_id: request_id.to_string(),
-                        details: pending.details,
-                    }
-                };
-
-                self.emit_approval(event);
-                tracing::debug!(%request_id, %accepted, "Resolved approval request");
-            }
+            let event = if accepted {
+                ApprovalRequest::Resolved {
+                    request_id: request_id.to_string(),
+                    details,
+                }
+            } else {
+                ApprovalRequest::Rejected {
+                    request_id: request_id.to_string(),
+                    details,
+                }
+            };
+
+            self.emit_approval(event);
+            tracing::debug!(%request_id, %accepted, "Resolved approval request");
```

Comment:
I think there might be some race condition here. Check if you can actually accept the offers that are displayed on the deposit page . If not there is an issue here (I might have an idea what it is)